src/eos/adiabatic_glmmhd.hpp

```diff
@@ -52,6 +52,98 @@ class AdiabaticGLMMHDEOS : public EquationOfState {
   }
   //
 
+  //----------------------------------------------------------------------------------------
+  // \!fn Real EquationOfState::ConsToPrim(View4D cons, View4D prim, const int& k, const
+  // int& j, const int& i) \brief Fills an array of primitives given an array of
+  // conserveds, potentially updating the conserved with floors
+  template <typename View4D>
+  KOKKOS_INLINE_FUNCTION void ConsToPrim(View4D cons, View4D prim, const int &k,
+                                         const int &j, const int &i) const {
+    Real &u_d = cons(IDN, k, j, i);
+    Real &u_m1 = cons(IM1, k, j, i);
+    Real &u_m2 = cons(IM2, k, j, i);
+    Real &u_m3 = cons(IM3, k, j, i);
+    Real &u_e = cons(IEN, k, j, i);
+    Real &u_b1 = cons(IB1, k, j, i);
+    Real &u_b2 = cons(IB2, k, j, i);
+    Real &u_b3 = cons(IB3, k, j, i);
+    Real &u_psi = cons(IPS, k, j, i);
+
+    Real &w_d = prim(IDN, k, j, i);
+    Real &w_vx = prim(IV1, k, j, i);
+    Real &w_vy = prim(IV2, k, j, i);
+    Real &w_vz = prim(IV3, k, j, i);
+    Real &w_p = prim(IPR, k, j, i);
+    Real &w_Bx = prim(IB1, k, j, i);
+    Real &w_By = prim(IB2, k, j, i);
+    Real &w_Bz = prim(IB3, k, j, i);
+    Real &w_psi = prim(IPS, k, j, i);
+
+    // apply density floor, without changing momentum or energy
+    u_d = (u_d > density_floor_) ? u_d : density_floor_;
+    w_d = u_d;
+
+    Real di = 1.0 / u_d;
+    w_vx = u_m1 * di;
+    w_vy = u_m2 * di;
+    w_vz = u_m3 * di;
+
+    w_Bx = u_b1;
+    w_By = u_b2;
+    w_Bz = u_b3;
+    w_psi = u_psi;
+
+    Real e_k = 0.5 * di * (SQR(u_m1) + SQR(u_m2) + SQR(u_m3));
+    Real e_B = 0.5 * (SQR(u_b1) + SQR(u_b2) + SQR(u_b3));
+    Real gm1 = gamma_ - 1.0;
+    w_p = gm1 * (u_e - e_k - e_B);
+
+    // apply pressure floor, correct total energy
+    u_e = (w_p > pressure_floor_) ? u_e : ((pressure_floor_ / gm1) + e_k + e_B);
+    w_p = (w_p > pressure_floor_) ? w_p : pressure_floor_;
+  }
+
+  //----------------------------------------------------------------------------------------
+  // \!fn Real EquationOfState::PrimToCons(View4D prim, View4D cons, const int& k, const
+  // int& j, const int& i) \brief Fills an array of conserveds given an array of
+  // primitives,
+  template <typename View4D>
+  KOKKOS_INLINE_FUNCTION void PrimToCons(View4D prim, View4D cons, const int &k,
+                                         const int &j, const int &i) const {
```

Comment:
Before I forget this issue, these functions do not enforce floors. Should this function also modify `prim` to enforce floors like `ConsToPrim` can modify `cons`? Probably?

Reply:
Actually, I ended up not using these functions at all. I will remove them for now